src/ioctls/vcpu.rs

```diff
@@ -928,6 +928,93 @@ mod tests {
         }
     }
 
+    #[cfg(target_arch = "aarch64")]
+    #[test]
+    fn test_run_code() {
```

Comment:
We can create a single test_run_code for both platforms to reduce the code duplication.
I don't know if it is possible, but ideally we would trigger the same exits on both x86 and arm with same values for reads and writes so we can have the same asserts in the exits. This would make the run loop to look the same for both platforms. Then the only differences between the two platforms would be:
- the actual code (we would have a aarch64 code and a x86_64 code)
- the register setup (sregs and regs for x86 and one_reg for arm)

Reply:
> We can create a single test_run_code for both platforms to reduce the code duplication.

There's a bit of overlap, like creating a VM with a memory region and putting guest code named 'code' (which is of course different) there. But then that's it.

> I don't know if it is possible, but ideally we would trigger the same exits on both x86 and arm with same values for reads and writes so we can have the same asserts in the exits.

Only the MMIO exits can be the same. AArch64 doesn't have PortIO nor Hlt.

> This would make the run loop to look the same for both platforms. Then the only differences between the two platforms would be:
> 
>     * the actual code (we would have a aarch64 code and a x86_64 code)
> 
>     * the register setup (sregs and regs for x86 and one_reg for arm)

Besides these two things and the exits, the vcpu creation is even different. AArch64 has a funny little dance with the preferred target and a vcpu-init must be done too. I really don't think there's any value in trying to share this function between x86 and AArch64. In fact, I would argue the code savings would be nil after adding all the conditions.

Reply:
> Would you like to add the code with some description to the highlevel documentation as well? 

Sure, I'll take a look at that and also attempt the refactoring of common vm/vcpu setup code to see if we can get a bit more code sharing without introducing too many arch annotations.